parallel/tutorial.md

```diff
@@ -38,111 +38,121 @@ threads can be used simultaneously to do the computation, which comprises the se
+```bash
+gz physics -o unthrottled0
+# Simulate with island threading:
+gz physics -o unthrottled1
+# Simulate with island threading:
+gz physics -o unthrottled2
+...
+# Simulate with threaded position error correction:
+gz physics -o split_unthrottled0
+# Simulate with both types of threading
+gz physics -o split_unthrottled1
+gz physics -o split_unthrottled2
+...
+```
 
-* `island_threads`: integer number of threads to use for island threading
-* `thread_position_correction`: flag to turn threading on for ODE quickstep position error correction
+#### Other resources
 
-These flags have been added to [physics profiles](http://gazebosim.org/tutorials?tut=preset_manager&cat=physics)
-in several world files on the `diagnostics_scpeters` branch of gazebo:
+We provide some scripts to run an experiments to track the `real time factor`.
 
-* [test/worlds/revolute\_joint\_test.world](https://github.com/osrf/gazebo/blob/diagnostics_scpeters/test/worlds/revolute_joint_test.world#L12)
-* [worlds/pr2.world](https://github.com/osrf/gazebo/blob/diagnostics_scpeters/worlds/pr2.world#L12)
-* [worlds/dual_pr2.world](https://github.com/osrf/gazebo/blob/diagnostics_scpeters/worlds/dual_pr2.world#L12)
+Create a new folder:
 
-Simulate two pr2 robots without threading:
+```bash
+mkdir /tmp/gazebo_parallel
+cd /tmp/gazebo_parallel
+```
 
-~~~
-. ${WS}/devel_isolated/setup.bash
-gazebo --verbose -o unthrottled0 \
-  ${WS}/src/gazebo/worlds/dual_pr2.world
-~~~
+##### Pendulums
 
-Simulate two pr2 robots with island threading:
+```bash
+wget https://raw.githubusercontent.com/osrf/gazebo_tutorials/master/parallel/files/run_pendulum_tests.bash
+bash run_pendulums_test.bash
+```
 
-~~~
-. ${WS}/devel_isolated/setup.bash
-gazebo --verbose -o unthrottled2 \
-  ${WS}/src/gazebo/worlds/dual_pr2.world
-~~~
+##### PR2
 
-Simulate two pr2 robots with threaded position error correction:
+```bash
+wget https://raw.githubusercontent.com/osrf/gazebo_tutorials/master/parallel/files/run_pr2_tests.bash
+bash run_pr2_test.bash
+```
+##### Dual PR2
 
-~~~
-. ${WS}/devel_isolated/setup.bash
-gazebo --verbose -o split_unthrottled0 \
-  ${WS}/src/gazebo/worlds/dual_pr2.world
-~~~
+```bash
+wget https://raw.githubusercontent.com/osrf/gazebo_tutorials/master/parallel/files/run_dual_pr2_tests.bash
+bash run_dual_pr2_test.bash
+```
 
-Simulate two pr2 robots with both types of threading:
+##### Visualizing the data
 
-~~~
-. ${WS}/devel_isolated/setup.bash
-gazebo --verbose -o split_unthrottled2 \
-  ${WS}/src/gazebo/worlds/dual_pr2.world
-~~~
+```bash
+wget https://raw.githubusercontent.com/osrf/gazebo_tutorials/master/parallel/files/show_parallel_results.py
+python3 show_parallel_results.py revolute_joint_test/
+python3 show_parallel_results.py pr2/
+python3 show_parallel_results.py dual_pr2/
+```
+
+If you have a look to the results obtained for the pr2. You will see that threaded islands don’t help the single PR2 scenario since the complex PR2
```

Comment:
How about adding some plots to the tutorial so users have a reference and know what to expect?

Reply:
Thanks for adding the plot from the PDF. What I was mainly referring to here were the plots generated by your new scripts. Do you think you could add them?

For example, I got this with the `revolute_joint_test`, which shows a significant increase in RTF as threads are added. (I wouldn't add my exact graph though because I didn't run it in a dedicated system)

![threaded](https://user-images.githubusercontent.com/5751272/83576038-ae7b3e80-a4e5-11ea-8a14-85f64c2b38c8.png)